File: programming files/rgb_to_hex.py
# this method takes in three values, r, g, b for red, green, and blue
def rgb_to_hex(r, g, b):
    # each variable gives the min and max color values for each red, green, and blue
    r = max(0, min(255, r))
    g = max(0, min(255, g))
    b = max(0, min(255, b))
    # returns each of the values formatted
    return '{:02X}{:02X}{:02X}'.format(r, g, b)


# test with hex_color = rgb_to_hex(255, 127, 0) # returns "FF7F00"

File: programming files/rgb_to_hex.java
public class rgb_to_hex {
    public static void main(String[] args) {
        // define the RGB values
        int r = 255;    // red component
        int g = 127;    // green component
        int b = 0;        // blue component

        // convert the RGB to the hexadecimal
        String hexColor = rgbToHex(r, g, b);

        // display the RGB and hexadecimal color values  
        System.out.println("RGB color (" + r + ", " + g + ", " + b + ") = " + hexColor);
    }

    public static String rgbToHex(int r, int g, int b) {

        // making sure that the values are within the valid range of 0-255 
        r = Math.min(255, Math.max(0, r));
        g = Math.min(255, Math.max(0, g));
        b = Math.min(255, Math.max(0, b));

        // convert the RBG to a hexadecimal string and format with leading zeros
        return String.format("%02X%02X%02X", r, g, b);
    }
}

//Test with RGB color (255, 127, 0) = FF7F00


